### PR TITLE
- Include only the parts from the storageframework that you really need.

### PR DESCRIPTION
**Pull request by baldersheim**
- Avoid default values to virtual method registerTick.

@vekterli PR

### CODE REVIEW

File: storage/src/vespa/storageframework/generic/thread/threadpool.h [hunk 1]
```diff
@@ -31,11 +31,8 @@ struct ThreadVisitor {
 struct ThreadPool {
     virtual ~ThreadPool() = default;
 
-    virtual Thread::UP startThread(Runnable&,
-                                   vespalib::stringref id,
-                                   vespalib::duration waitTime,
-                                   vespalib::duration maxProcessTime,
-                                   int ticksBeforeWait,
+    virtual std::unique_ptr<Thread> startThread(Runnable&, vespalib::stringref id, vespalib::duration waitTime,
+                                   vespalib::duration maxProcessTime, int ticksBeforeWait,
```
Comment (vekterli) [inline storage/src/vespa/storageframework/generic/thread/threadpool.h:35]: Nit: indentation out of sync

**Review by vekterli — APPROVED** [review]

File: storage/src/tests/visiting/visitormanagertest.cpp [hunk 2]
```diff
@@ -312,11 +312,9 @@ VisitorManagerTest::verifyCreateVisitorReply(
 uint32_t
 VisitorManagerTest::getMatchingDocuments(std::vector<document::Document::SP >& docs) {
     uint32_t equalCount = 0;
-    for (uint32_t i=0; i<docs.size(); ++i) {
-        for (uint32_t j=0; j<_documents.size(); ++j) {
-            if (docs[i]->getId() == _documents[j]->getId()
-                && *docs[i] == *_documents[j])
-            {
+    for (auto & doc : docs) {
+        for (auto & _document : _documents) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/tests/visiting/visitormanagertest.cpp]: Fixed

File: storage/src/vespa/storage/visiting/visitormanager.cpp [hunk 3]
```diff
@@ -653,9 +654,9 @@ VisitorManager::reportHtmlStatus(std::ostream& out,
     std::sort(_statusRequest.begin(), _statusRequest.end(), StatusReqSorter());
 
     // Create output
-    for (uint32_t i=0; i<_statusRequest.size(); ++i) {
-        out << "<h2>" << _statusRequest[i]->getSortToken()
-            << "</h2>\n" << _statusRequest[i]->getStatus() << "\n";
+    for (auto & request : _statusRequest) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/visiting/visitormanager.cpp]: Fixed

File: storage/src/tests/visiting/visitormanagertest.cpp [hunk 4]
```diff
@@ -47,10 +47,10 @@ struct VisitorManagerTest : Test {
     std::unique_ptr<TestVisitorMessageSessionFactory> _messageSessionFactory;
     std::unique_ptr<TestServiceLayerApp> _node;
     std::unique_ptr<DummyStorageLink> _top;
-    VisitorManager* _manager;
+    VisitorManager* _manager{};
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/tests/visiting/visitormanagertest.cpp]: Fixed

**Pull request merged**